src/main/java/hudson/tasks/junit/CaseResult.java

```diff
@@ -84,6 +84,8 @@ public class CaseResult extends TestResult implements Comparable<CaseResult> {
     private String errorStackTrace;
     private String errorDetails;
     private Map<String, String> properties;
+    private FlakyFailure[] flakyFailures;
```

Comment:
any reason these are an array rather than a list?

Reply:
for some reasons, using `List` is creating some empty elements in xml and fail `TestResultTest#bigResultReadWrite`

Reply:
Because you are using `record`s, I think. XStream does not support them well. https://github.com/x-stream/xstream/issues/345 If you switch to `final class`es (yes it is irritating) then you should be able to make this a `List<FlakyFailure>` and XStream should produce (semi-)attractive and functional XML.

Just be sure the concrete type is `ArrayList`! Do not try to use `List.of()`, `Collections.unmodifiableList`, etc. in the field itself. You can always clone lists in a getter and/or setter or use an unmodifiable collection in a getter.

Reply:
The problem was more about returning an empty liist which create an empty element in the xml and so was breaking the test `TestResultTest#bigResultReadWrite`
Now by using List but returning null instead of empty list the test is passing.

Reply:
Yeah just double-check the XML format on disk by inspection.